### PR TITLE
Tech Story: [M3-7947] - Update storybook to 8.0.5

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
PR to update storybook to `8.0.5`
- See [release notes](https://storybook.js.org/blog/storybook-8/)
- See [migration guide](https://storybook.js.org/docs/migration-guide#missing-%20%20%20%E2%94%82%20%E2%94%82%20%20%20viteconfigjs-file)

I did not change anything to how we're type-checking or compiling. This could be done in subsequent optimizing PRs. It should be a 1/1 update to prepare us to adopt some extra features down the line.

I took a brief stab at adopting `react-docgen` (much faster, instead of `react-docgen-typescript`) by default since it is much faster, but type fidelity benefits were lost so best to keep the current config for now.

## Changes  🔄
- Update CM storybook to `8.0.5`
- Update other relevant packages
- Update relevant imports
- Update last remaining `.mdx` file

## Preview 📷
There should be no degradation or regression to our storybook as a result of this PR

### Verification steps
- Pull PR locally
- Run `yarn storybook` and browse stories to confirm no regressions
- Run `yarn build-storybook` and `storybook-static` to confirm build will succeeds in our pipelines

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by bnussman-akamai — APPROVED** [review]
Awesome. Thank you 🎉 

- Confirmed local dev works
- Confirmed build works
- Confirmed `react-docgen-typescript` generates better docs than `react-docgen`
- Confirmed stories show and render as expected
- Confirmed dark mode toggle works as expected

File: packages/manager/src/components/intro.mdx [hunk 1]
```diff
@@ -1,13 +1,13 @@
-import { Meta } from '@storybook/addon-docs';
+import { Meta } from '@storybook/blocks';
 
 <Meta title="Intro/Welcome" />
 
 # Welcome
 
-_The mission of **Linode** is to accelerate innovation by making cloud computing simple, affordable, and accessible to all._
+*The mission of **Linode** is to accelerate innovation by making cloud computing simple, affordable, and accessible to all.*
```
Comment (mjac0bs) [inline packages/manager/src/components/intro.mdx:7]: It feels/looks kind of weird to italicize this sentence - at least to me, it makes it harder to read - but that's beside the point of this PR.

File: packages/manager/.changeset/pr-10336-tech-stories-1711993874664.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Tech Stories
+---
+
+Update storybook to 8.0.5 ([#10336](https://github.com/linode/manager/pull/10336))
```
Comment (mjac0bs) [inline packages/manager/.changeset/pr-10336-tech-stories-1711993874664.md:5]:
```suggestion
Update Storybook to 8.0.5 ([#10336](https://github.com/linode/manager/pull/10336))
```
Nit.

**Review by mjac0bs — APPROVED** [review]
Thank you!  📖 🚀  Build steps ran successfully, Storybook started successfully locally and from browsing various pages and adjusting props in the playgrounds, I did not see any regressions.

Screenshot 2024-04-01 at 2 24 15 PM
Screenshot 2024-04-01 at 2 24 57 PM

File: packages/manager/src/components/intro.mdx (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/src/components/intro.mdx]: You're right, i changed it, why not

File: packages/manager/.changeset/pr-10336-tech-stories-1711993874664.md (hunk 2 above)
Comment (abailly-akamai) [reply packages/manager/.changeset/pr-10336-tech-stories-1711993874664.md]: yup, done

**Pull request merged**